packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -146,10 +155,259 @@ class WidgetInspectorService {
 
   List<String> _pubRootDirectories;
 
+  /// Registers a service extension method with the given name (full
+  /// name "ext.flutter.inspector.name").
+  ///
+  /// The given callback is called when the extension method is called. The
+  /// callback must return a value that can be converted to JSON using
+  /// `json.encode()` (see [JsonEncoder]). The return value is stored as a
+  /// property named `result` in the JSON. In case of failure, the failure is
+  /// reported to the remote caller and is dumped to the logs.
+  @protected
+  void registerServiceExtension({
```

Comment:
isn't this duplicating a lot of code from the binding layer?

Reply:
two methods are duplicated. The others support the specific API signature I've found convenient for the inspector of methods accepting two string valued arguments the first being an object id and the second an object group. cleaned up the most egregious duplicate code as discussed bellow.